doc/manual/manual.tex

```diff
@@ -5,183 +5,241 @@
+paths being wrong have been reported. If you face such issues please look at 
+Section~\ref{sec:troubleshooting}.
 
-Command-line arguments are passed to ALE before the ROM filename. TODO: Provide a different
-example to use the command-line arguments (if any, since we removed the internal agent).
-%: For example, assuming you have the appropriate ROM file saved in the directory 
-%\verb+ale_0_4/roms+, the following runs the internal random agent on Space Invaders:
+\section{Implementing an agent in C++ (shared libraries)}
 
-%\begin{verbatim}
-%ale_0_4> ./ale -game_controller internal -player_agent random_agent  
-%  roms/space_invaders.bin
-%\end{verbatim}
+If one wants to implement an agent in C++, the best approach is to use the ALE shared library. 
+The shared library interface allows agents to directly access ALE via a class called
+\verb+ALEInterface+, defined in \verb+ale_interface.hpp+. We are going to discuss here, 
+step-by-step how to implement an agent that plays randomly. A code containing an example is
+available at \verb+doc/examples/sharedLibraryInterfaceExample.cpp+.
 
-The configuration file \verb+ale_0_4/stellarc+ can also be used to set frequently used
-command-line arguments. 
+This example is automatically compiled with the ALE when following the steps presented in
+Section~\ref{sec:install}. If one wants to disable such an option it is enough to replace \verb+ON+ 
+by \verb+OFF+ in the line below, present in file \verb+ale_0_5/CMakeLists.txt+:
 
-\subsection{Main Arguments}
-\small{
 \begin{verbatim}
+  option(BUILD_EXAMPLES "Build Example Agents" ON)
+\end{verbatim}
 
-  -help -- prints out help information
+To implement an agent, the first step is to include the library \verb+ale_interface.hpp+, either via 
+the relative path \verb+#include ``path/from/your/code/ale_interface.hpp''+, or as a standard 
+header: \verb+#include <ale_interface.hpp>+. If the later is chosen, remember to add the proper 
+path using the flag \verb+-I+ when compiling the code.
 
-  -game_controller <fifo|fifo_named|rlglue> -- selects an ALE interface
-    default: unset
+To instantiate the Arcade Learning Environment it is enough to do:\\
 
-  -random_seed <###|time> -- picks the ALE random seed, or sets it to current time
-    default: time
+\verb+ALEInterface ale;+\\
 
-  -display_screen <true|false> -- if true and SDL is enabled, displays ALE screen
-    default: false
+Once the environment is initialized, it is now possible to set its arguments. This is done with the 
+functions \verb+setBool(), setInt(), setFloat()+. The complete list of flags is available in 
+Section~\ref{sec:arguments}. Just as an example, if one wants to set the environment's seed he must 
+do:\\
 
-\end{verbatim}
-}
+\verb+ale.setInt("random_seed", 123);+\\
 
-\subsection{Environment Arguments}
+After setting all variables we can now load the game ROM:\\
 
-\small{
-\begin{verbatim}
+\verb+ale.loadROM(asterix.bin /*rom name*/);+\\
```

Comment:
In-line comment is a bit weird. Why not put after the semicolon with standard //?